src/Sylius/Component/Core/Promotion/Checker/NthProductRuleChecker.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Sylius\Component\Core\Promotion\Checker;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Promotion\Checker\RuleCheckerInterface;
+use Sylius\Component\Promotion\Exception\UnsupportedTypeException;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Symfony\Component\VarDumper\VarDumper;
+
+/**
+ * checks if order contains nth times the same product
+ */
+class NthProductRuleChecker implements RuleCheckerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isEligible(PromotionSubjectInterface $subject, array $configuration)
+    {
+        if (!$subject instanceof OrderInterface) {
+            throw new UnsupportedTypeException($subject, 'Sylius\Component\Core\Model\OrderItemInterface');
+        }
+
+        foreach ($subject->getItems() as $item) {
+            if ($item->getQuantity() >= $configuration['nth']) {
+                return true;
+            }
```

Comment:
I didn't quite get the logic here. If any of the items has a quantity more than what's configured the rule applies?

So if I order 10 different products with 1 quantity I'm not going to get a promotion of `nth promotion with 2 as config`?